### DISCUSSION TITLE
LogGroups should be added as CloudFormation resources

### DISCUSSION THREAD

**Discussion by estahn** [serverless/serverless#11175] [General]
1. We ran a script to automatically cleanup log streams and empty log groups.
2. We ran cloudformation drift detection and run a script to automatically remediate drift.
3. We ran `sls deploy` which failed due to missing log groups.

Immo this should have been picked up by drift detection / remediation.

---
**UPDATE: CloudFormation drift does not work for LogGroups. This is unrelated to serverless.**

---

**Comment by pgrzesik** [comment]
Hello @estahn - could you please include the output of the `sls deploy` that you've run? Why exactly did you remove the empty log groups manually?

**Comment by estahn** [comment]
@pgrzesik 

**Output:**
```
Deploying xxx-data-pipeline to stage dev (ap-southeast-2, "arn:aws:iam::xxx:role/SFRole-xxx" provider)
--
377 |  
378 | × Stack xxx-data-pipeline-dev failed to deploy (261s)
379 | Environment: linux, node 14.15.4, framework 3.18.2 (local), plugin 6.2.2, SDK 4.3.2
380 | Credentials: Serverless Dashboard, "arn:aws:iam::xxx:role/SFRole-xxx" provider (https://app.serverless.com/xxx/apps/xxx-data-pipeline/xxx-data-pipeline/dev/ap-southeast-2/providers)
381 | Docs:        docs.serverless.com
382 | Support:     forum.serverless.com
383 | Bugs:        github.com/serverless/serverless/issues
384 |  
385 | Error:
386 | UPDATE_FAILED: EnterpriseLogAccessIamRole (AWS::IAM::Role)
387 | Unable to retrieve Arn attribute for AWS::Logs::LogGroup, with error message Resource of type 'AWS::Logs::LogGroup' with identifier '{"/properties/LogGroupName":"/aws/lambda/xxx-data-pipeline-dev-foobarxxx"}' was not found.
```

We had a lot of stale log streams and log groups collected over the years, so we were running a script to remove empty log streams and log groups.

**Reply by pgrzesik** [reply]
Thanks for sharing, and what script did you run to remediate the drift? Something custom-made?

**Reply by estahn** [reply]
https://github.com/WeAreCloudar/cfn-drift-remediation

Was thinking of moving this into a serverless plugin. Just finishing up another plugin.

Would be good to get something along the lines of:
`sls drift show`
`sls drift remediate`

etc

**Reply by pgrzesik** [reply]
Thanks for sharing. And what about the actual existence of the referenced log group - was it recreated or not?

**Reply by estahn** [reply]
What I found out is that CloudFormation drift is not detecting deleted LogGroups. Not sure if that is a bug or expected behaviour.

**Reply by pgrzesik** [reply]
That's interesting, if they're a part of CF template I think they should be recreated. Did you have a chance to contact AWS about this behavior?

**Comment by estahn** [comment]
Actually I might be wrong and those are already added to Cloudformation 🤔
